src/analyze/analyze-condition.c

```diff
@@ -72,29 +73,57 @@ static int log_helper(void *userdata, int level, int error, const char *file, in
         return r;
 }
 
-int verify_conditions(char **lines, UnitFileScope scope) {
+int verify_conditions(char **lines, UnitFileScope scope, const char *unit, const char *root) {
         _cleanup_(manager_freep) Manager *m = NULL;
         Unit *u;
-        char **line;
         int r, q = 1;
 
+        if (unit) {
+                _cleanup_strv_free_ char **filenames = NULL;
+                _cleanup_free_ char *var = NULL;
+
+                r = strv_extend(&filenames, unit);
+                if (r < 0)
+                        return r;
```

Comment:
```c
filenames = strv_new(unit);
```

(i mean, sure, extending an empty list is the same as allocating a new list more or less, by conceptually it's easier to grok to just write this as: allocate a new list)